kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/io/kotest/datatest/styles/StringSpecDataTest.kt

```diff
@@ -0,0 +1,26 @@
+package io.kotest.datatest.styles
+
+import io.kotest.core.names.DuplicateTestNameMode
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.datatest.assertDataTestResults
+import io.kotest.datatest.registerRootTests
+import io.kotest.matchers.shouldBe
+
+class StringSpecDataTest : StringSpec() {
+   init {
+
+      duplicateTestNameMode = DuplicateTestNameMode.Silent
+
+      val results = registerRootTests()
+      var count = 0
+
+      afterTest {
+         count++
+      }
+
+      afterSpec {
+         results.assertDataTestResults()
+         count shouldBe 36
```

Comment:
will this `36` be fragile, change every time we change `registerRootTests`? If yes, is there a way to match to some number computed off of `registerRootTests`?

Reply:
I don't think we change these tests often enough for it to be a worry